### PR TITLE
feat: Types for html-to-pdfmake

### PR DESCRIPTION
**Pull request by mzl-md**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by mzl-md** [comment]
Hi @aymkdn, could you please have a look at this? Are you ok with me adding you as an owner?

**Comment by Aymkdn** [comment]
Hi @mzl-md 

Sorry but I'm not sure to understand what is this repo and what you're expecting from me? I'd appreciate to have some explanations.

Thanks

**Comment by mzl-md** [comment]
This repo contains typescript type definitions for javascript libraries that don't provide some by themself. We are using [html-to-pdfmake](https://github.com/Aymkdn/html-to-pdfmake) in a typescript project and would like to add suitable type definitions here so others can use them too.
I created the [index.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52470/files#diff-dc6ecaba854c632683d3f6f3676bf12a64f914de8fe517027a635bebccdf3444) from your readme file, but I'm not sure if the types that I imported from pdfmake really are pdfmake stuff or something else?

**Comment by Aymkdn** [comment]
I never worked with typescript, so I don't know what I have to check…

In [index.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52470/files#diff-dc6ecaba854c632683d3f6f3676bf12a64f914de8fe517027a635bebccdf3444) I can see the declarations of the parameters for my lib… But that's it?! I don't know if this file is supposed to contain something else.

Sorry, but my knowledge here is very limited.

**Comment by mzl-md** [comment]
The declaration is everything that is required. It would be great if you could confirm that the parameters where I used imports from pdfmake are supposed to be pdfmake types and not something html-to-pdfmake internal.

**Comment by Aymkdn** [comment]
All the below code is related to `html-to-pdfmake`:
```ts
declare function htmlToPdfmake(
    html: string,
    options?: Partial<{
        /**
         * You can overwrite the default styles using defaultStyles
         */
        defaultStyles: import("pdfmake/interfaces").StyleDictionary;
        /**
         * You can overwrite the default sizes for the old HTML4 tag <font> by using fontSizes. It must be an array with 7 values
         */
        fontSizes: [number, number, number, number, number, number, number];
        /**
         * By passing replaceText as a function with two parameters (text and nodes) you can modify the text of all the nodes in your HTML document.
         */
        replaceText: (text: string, nodes: Array<import("pdfmake/interfaces").Node>) => string;
        /**
         * By passing tableAutoSize with true, then the program will try to define widths and heights for the tables, based on CSS properties width and height that have been provided to TH or TD.
         */
        tableAutoSize: boolean;
        /**
         * If you use Node, then you'll have to pass the window object
         */
        window: unknown;
    }>,
)
```

`defaultStyles`, `fontSizes`, `replaceText`, `tableAutoSize` and `window`  are `html-to-pdfmake` only parameters.

I don't know if I answered your question?

File: types/html-to-pdfmake/index.d.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ declare function htmlToPdfmake(
     html: string,
     options?: Partial<{
         /**
-         * You can overwrite the default styles using defaultStyles
+         * You can overwrite the default styles using defaultStyles.
+         * Please, note that the above default styles are stronger than the ones defined in the style classes.
          */
         defaultStyles: import("pdfmake/interfaces").StyleDictionary;
```
Comment (Aymkdn) [inline types/html-to-pdfmake/index.d.ts:17]: I don't think this is correct… `defaultStyles` has nothing to do with `pdfmake`

Comment (mzl-md) [reply types/html-to-pdfmake/index.d.ts:17]: Looking at your sources I think so too. But which style attributes are allowed? Everything html allows?

Comment (Aymkdn) [reply types/html-to-pdfmake/index.d.ts:17]: > But which style attributes are allowed? Everything html allows?

Yes, with the camel case (e.g. "fontSize", "marginBottom", …)

Comment (mzl-md) [reply types/html-to-pdfmake/index.d.ts:17]: I'm a little confused. The example `defaultStyles` in your Readme contains `bold: true`, `decoration: "underline"`, and `italics: true`, which are not valid CSS attributes but pdfmake properties.

Comment (Aymkdn) [reply types/html-to-pdfmake/index.d.ts:17]: You're right… It's indeed the PDFMake style. Sorry for the confusion!

Comment (mzl-md) [reply types/html-to-pdfmake/index.d.ts:17]: Alright, I think the types are now correct. Thanks for your help!
Could you please approve the PR so the maintainers can see this PR is ready to merge?

**Review by Aymkdn — APPROVED** [review]

**Comment by Aymkdn** [comment]
Do you need anything else from me?

**Comment by mzl-md** [comment]
No I guess that's all. Thank again for your help!

**Review by orta — APPROVED** [review]
Looks good to me based on the npm README and your conversations

**Comment by mzl-md** [comment]
Ready to merge

**Pull request merged**